docs/src/faq.md

```diff
@@ -1,5 +1,31 @@
 # FAQ
 
+## Why do I get "missing field X" error when running `install`?
+
+This error indicates one of your dependencies has an invalid manifest file. The
+error message should indicate which dependency is broken and why:
+
+```
+Error:
+   0: Failed to process dependency foo-bar@0.0.1
+   1: Failed to parse the manifest
+   2: TOML parse error at line 1, column 1
+   2:   |
+   2: 1 | [package]
+   2:   | ^
+   2: missing field `dependencies`
+```
+
+Since the CLI won't allow you to publish a package with an invalid manifest,
+you may wonder why some published packages may be broken in this way. The
+reason is that all fields used to be optional -- `impl` packages didn't need a
+`package` section, and `lib` packages didn't require `dependencies`. When the
+lockfile functionality was first introduced, both sections were made mandatory
+for all package types, which broke parsing of older manifests.
```

Comment:
Yes, that's right. I assume this is just to match Cargo — or was there another reason @mara-schulke?

Reply:
It is to have a strict structure for declaring manifests (which always consists of [package] + [dependencies]) – i thought this would a) simplify the learning curve as someone doesn't need to know when which section is mandatory and when not.

This creates symmetry between libraries, apis and implementations (+ im somewhat confident that we could enable lib to lib deps in some distant future)

Reply:
But Cargo _doesn't_ require `[dependencies]` if it's empty.

I guess what I'm trying to understand is the value of always including `[dependencies]` — it doesn't feel like something that aids learning, because now you have to put a thing in a file even if you don't need said thing.

Now, it'd be different if you asserted that every project will have at least one dependency, but I don't think that's true?

Reply:
Okay agreed, making the package section mandatory was the important change - the dependency section can be sensibly defaulted to an empty list

Reply:
Made the change.